samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -104,14 +116,29 @@ public synchronized void stop() {
       coordinatorListener.onJobModelExpired();
     }
 
+    metrics.isLeader.set(0);
```

Comment:
Added the comment. We are setting the isLeader metric to false here because when the stream processor shuts down, it does not remain the leader anymore